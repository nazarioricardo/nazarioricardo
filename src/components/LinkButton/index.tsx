import Link, { LinkProps } from "next/link";
import { HTMLProps, FC, RefAttributes } from "react";
import { OverridableComponent } from "@mui/material/OverridableComponent";
import { SvgIconTypeMap } from "@mui/material";
import styles from "./styles.module.css";

type LinkButtonProps = {
  text: string;
  Icon?: OverridableComponent<SvgIconTypeMap<{}, "svg">> & { muiName: string };
  gradient?: boolean;
} & LinkProps &
  RefAttributes<HTMLAnchorElement>;

function LinkButton({ text, Icon, href, gradient, ...props }: LinkButtonProps) {
  return (
    <Link
      {...props}
      href={href}
      className={`${styles.linkButton} ${gradient && styles.gradient}`}
    >
      {text} {Icon && <Icon />}
    </Link>
  );
}

export default LinkButton;
