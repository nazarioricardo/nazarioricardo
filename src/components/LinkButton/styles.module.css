.linkButton {
  display: flex;
  flex-direction: row;
  justify-content: center;
  gap: 12px;
  align-items: center;
  background-color: rgb(var(--background-rgb));
  font-family: var(--font-josefin-sans);
  font-size: 1.1rem;
  min-height: 52px;
  border: none;
  border-radius: 26px;
  padding-left: 26px;
  padding-right: 26px;
  color: rgb(var(--foreground-rgb));
  cursor: pointer;
  text-align: center;
}

.linkButton:hover {
  background-color: rgb(var(--hover-rgb));
}

@media only screen and (max-width: 1170px) {
  .linkButton {
    min-height: 48px;
    border-radius: 24px;
    padding-left: 24px;
    padding-right: 24px;
  }
}
