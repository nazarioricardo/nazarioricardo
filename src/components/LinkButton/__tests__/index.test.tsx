import React from "react";
import { render, screen, fireEvent } from "@testing-library/react";
import "@testing-library/jest-dom";
import LinkButton from "../index"; // adjust this import according to your file structure
import { Home } from "@mui/icons-material";

describe("LinkButton", () => {
  test("renders LinkButton component", () => {
    render(<LinkButton text="Home" Icon={Home} href="/home" />);
    const linkElement = screen.getByRole("link", { name: /Home/ });
    expect(linkElement).toBeInTheDocument();
    expect(linkElement).toHaveAttribute("href", "/home");
  });

  test("renders LinkButton component with icon", () => {
    render(<LinkButton text="Home" Icon={Home} href="/home" />);
    const iconElement = screen.getByTestId("HomeIcon");
    expect(iconElement).toBeInTheDocument();
  });

  test("navigates to the correct page when link is clicked", () => {
    render(<LinkButton text="Home" Icon={Home} href="/home" />);
    const linkElement = screen.getByRole("link", { name: /Home/ });
    fireEvent.click(linkElement);
  });

  test("matches snapshot", () => {
    const { asFragment } = render(
      <LinkButton text="Home" Icon={Home} href="/home" />
    );
    expect(asFragment()).toMatchSnapshot();
  });

  // Add more tests as needed
});
