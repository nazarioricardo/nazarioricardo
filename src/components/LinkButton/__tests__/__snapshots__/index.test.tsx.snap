// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkButton matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="linkButton undefined"
    href="/home"
  >
    Home 
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
      data-testid="HomeIcon"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
      />
    </svg>
  </a>
</DocumentFragment>
`;
